src/3d/qgs3dmapscene.cpp

```diff
@@ -1218,35 +1221,34 @@ QVector<const QgsChunkNode *> Qgs3DMapScene::getLayerActiveChunkNodes( QgsMapLay
 
 QgsRectangle Qgs3DMapScene::sceneExtent()
 {
-  QgsRectangle extent;
-  extent.setMinimal();
+  return mMap.extent();
+}
 
-  for ( auto it = mLayerEntities.constBegin(); it != mLayerEntities.constEnd(); it++ )
+QgsDoubleRange Qgs3DMapScene::elevationRange() const
+{
+  double yMin = std::numeric_limits< double >::max();
+  double yMax = std::numeric_limits< double >::lowest();
+  if ( mMap.terrainRenderingEnabled() && mTerrain )
   {
-    QgsMapLayer *layer = it.key();
-    Qt3DCore::QEntity *layerEntity = it.value();
-    QgsChunkedEntity *c = qobject_cast<QgsChunkedEntity *>( layerEntity );
-    if ( !c )
-      continue;
-    QgsChunkNode *chunkNode = c->rootNode();
-    QgsAABB bbox = chunkNode->bbox();
-    QgsRectangle layerExtent = Qgs3DUtils::worldToLayerExtent( bbox, layer->crs(), mMap.origin(), mMap.crs(), mMap.transformContext() );
-    extent.combineExtentWith( layerExtent );
+    const QgsAABB bbox = mTerrain->rootNode()->bbox();
+    yMin = std::min( yMin, ( double )bbox.yMin );
+    yMax = std::max( yMax, ( double )bbox.yMax );
   }
 
-  if ( mMap.terrainRenderingEnabled() )
+  for ( auto it = mLayerEntities.constBegin(); it != mLayerEntities.constEnd(); it++ )
   {
-    if ( QgsTerrainGenerator *terrainGenerator = mMap.terrainGenerator() )
+    QgsMapLayer *layer = it.key();
+    if ( layer->type() == QgsMapLayerType::PointCloudLayer )
     {
-      QgsRectangle terrainExtent = terrainGenerator->rootChunkExtent();
-      QgsCoordinateTransform terrainToMapTransform( terrainGenerator->crs(), mMap.crs(), QgsProject::instance() );
-      terrainToMapTransform.setBallparkTransformsAreAppropriate( true );
-      terrainExtent = terrainToMapTransform.transformBoundingBox( terrainExtent );
-      extent.combineExtentWith( terrainExtent );
+      QgsPointCloudLayer *pcl = qobject_cast< QgsPointCloudLayer *>( layer );
+      QgsDoubleRange zRange = pcl->elevationProperties()->calculateZRange( pcl );
+      yMin = std::min( yMin, zRange.lower() );
+      yMax = std::max( yMax, zRange.upper() );
     }
```

Comment:
Why not call calculateZRange for all layer types (where elevationProperties is non null!), and ignore empty returned ranges? Then when other layer types gain implementations of calculateZRange this code will automatically handle them.

Reply:
This z range would be merely indicative of the actual y range in the 3d view, as the 3d geometries may have elevation relative to the terrain, be translated or have big or data defined height/extrusion parameters. That's why I decided to skip vector layers for this

Reply:
I'd suggest we add some flags to calculateZRange to allow the behaviour to be tweaked, eg a "AllowEstimates" flag could be used to indicate "just give me a ballpark range which is quick to calculate", and then if this flag isn't present we could iterate through the vector features and calculate the actual range.

But I'm happy to defer that till the need arises!